src/repositories_libraries_rust.tf

```diff
@@ -0,0 +1,97 @@
+  # {
+  #   owner_team = "services"
+  #   visibility = "public"
+  #   default_branch = "dev"
+  #   template = github_repository.svc_template_rust.name
+  #   default_branch_protection_settings = {} # Using module defaults
+  # }
+  lib_repos = {
+    #"router" = {
+    #  description = "Fleet Routing Algorithms"
+    #},
+    #"analysis" = {
+    #  description = "Performance Metrics for Fleet Routing"
+    #}
+  }
+
+  lib_files = {
+    "gitignore" = {
+      target   = ".gitignore",
+      location = "templates/rust-lib/.gitignore"
+    },
+    "editorconfig" = {
+      target   = ".editorconfig",
+      location = "templates/rust-all/.editorconfig"
+    },
+    "cargo-config" = {
+      target   = ".cargo/config.toml",
+      location = "templates/rust-all/.cargo/config.toml"
+    },
+    "cargo-husky-pre-commit" = {
+      target   = ".cargo-husky/hooks/pre-commit"
+      location = "templates/rust-all/.cargo-husky/hooks/pre-commit"
+    },
+    "cargo-husky-pre-push" = {
+      target   = ".cargo-husky/hooks/pre-push",
+      location = "templates/rust-all/.cargo-husky/hooks/pre-push"
+    },
+    "cargo-husky-hook-readme" = {
+      target   = ".cargo-husky/hooks/README.md",
+      location = "templates/rust-all/.cargo-husky/hooks/README.md"
+    },
+    "workflow-rust" = {
+      target   = ".github/workflows/rust_ci.yml",
+      location = "templates/rust-all/.github/workflows/rust_ci.yml"
+    },
+    "workflow-python" = {
+      target   = ".github/workflows/python_ci.yml",
+      location = "templates/rust-all/.github/workflows/python_ci.yml"
+    },
+    "workflow-editorconfig" = {
+      target   = ".github/workflows/editorconfig_check.yml",
+      location = "templates/rust-all/.github/workflows/editorconfig_check.yml"
+    }
+
+  }
+}
+
+module "lib_repository" {
+  source   = "./modules/github-repository/"
+  for_each = local.lib_repos
+
+  name        = format("lib-%s", each.key)
+  description = format("Arrow Library - %s", each.value.description)
+
+  # Settings with defaults
+  owner_team     = try(each.value.owner_team, "services")
+  visibility     = try(each.value.visibility, "public")
+  default_branch = try(each.value.default_branch, "develop")
+  template       = try(each.value.template, github_repository.lib_template_rust.name)
+
+  default_branch_protection_settings = try(each.value.default_branch_protection_settings, {})
+}
+
+########################################################
+#
+# Provision files
+#
+########################################################      
+resource "github_repository_file" "lib-file-provision" {
+  for_each            = local.lib_files
+  repository          = var.name
+  branch              = "main"
+  file                = each.value.target
+  content             = file(each.value.location)
+  commit_message      = "Provisioned by Terraform"
+  commit_email        = "automation@arrowair.com"
+  commit_author       = "Arrow automation"
+  overwrite_on_create = true
+}
```

Comment:
Move the entire files block to the `github-repository` module. This module will be called for each of the repositories, and these files, we want to provision for each repository as well.
https://github.com/Arrow-air/tf-github/blob/be36c3bfb190cb96f58edcfd6b25bdbd8eef415a/src/modules/github-repository/main.tf#L181
You have access to the repository name there as well